src/oph/ehoks/oppijaindex.clj

```diff
@@ -111,14 +111,15 @@
         "Opiskeluoikeus %s has multiple osaamisala. First one is used." oid))
     (let [tutkinto (get-tutkinto-nimi opiskeluoikeus)
           osaamisala (get-osaamisala-nimi opiskeluoikeus)]
-      {:oid oid
-       :oppija_oid oppija-oid
-       :oppilaitos_oid (get-in opiskeluoikeus [:oppilaitos :oid])
-       :koulutustoimija_oid (get-in opiskeluoikeus [:koulutustoimija :oid])
-       :tutkinto (:fi tutkinto)
-       :tutkinto_nimi tutkinto
-       :osaamisala (:fi osaamisala)
-       :osaamisala_nimi osaamisala})))
+      (-> {:oid oid
+           :oppija_oid oppija-oid
+           :oppilaitos_oid (get-in opiskeluoikeus [:oppilaitos :oid])
+           :koulutustoimija_oid (get-in opiskeluoikeus [:koulutustoimija :oid])
+           :tutkinto_nimi tutkinto
+           :osaamisala_nimi osaamisala}
+          (cond-> (some? (:fi tutkinto)) (assoc :tutkinto (:fi tutkinto)))
+          (cond-> (some? (:fi osaamisala))
```

Comment:
Muokkasin tätä vielä vähän. Huomasin että jos tutkinto ei ole null vaan siitä vaan puuttuu :fi esim näin {:sw "jotain ruotsia"} niin silloin (:fi tutkinto) palautti edelleen null ja sitten tallentaessa kosahtaisi.